src/helpers/amoc_xmpp_handlers.erl

```diff
@@ -57,12 +57,27 @@ stanza_handler(Pred, Action) ->
             end
     end.
 
+check_predicate(Pred, Stanza) when is_function(Pred, 1)->
+    apply_safely(Pred, [Stanza]).
+
 perform_action(Action, Client, Stanza, Metadata) when is_function(Action, 3) ->
-    Action(Client, Stanza, Metadata);
+    apply_safely(Action, [Client, Stanza, Metadata]);
 perform_action(Action, Client, Stanza, _Metadata) when is_function(Action, 2) ->
-    Action(Client, Stanza);
+    apply_safely(Action, [Client, Stanza]);
 perform_action(Action, _Client, _Stanza, _Metadata) when is_function(Action, 0) ->
-    Action().
+    apply_safely(Action, []).
+
+-spec apply_safely(function(), [term()]) -> {ok | error, term()}.
+apply_safely(F, A) ->
+    try erlang:apply(F, A) of
+        Result -> Result
+    catch
+        Class:Exception:Stacktrace ->
+            Error = {Class, Exception, Stacktrace},
+            ?LOG_ERROR("apply_safely crashed Fn = %p, Args = %p, Error = %p",
```

Comment:
Do we should using here `%p` instead of `~p`?

Reply:
damn my brain is damaged by almost a decade of embedded programming.